### ISSUE TITLE
Coverage not deterministic

### ISSUE THREAD

**Issue by bergwerf** [dart-lang/sdk#31658]
Coverage changes between runs (https://codecov.io/gh/qedb/expr/src/2e38745579abc262ede981aabf418a3c56cdacb6/lib/src/interface/printer.dart#L103, https://codecov.io/gh/qedb/expr/src/65fe7dfbbf51ef382de399db85a63ff8daba635a/lib/src/interface/printer.dart#L103). More examples can be found in this repository. The issue can be easily replicated with the observatory (I tried with `2.0.0-dev.8.0`). In all cases the line in question is definitely executed since the error message is matched in the test cases. I think this is related to #29330 which was a while ago. In all cases I can remember it concerns a `throw new X`. Perhaps some optimization is messing with coverage data?

**Comment by zoechi** [comment]
I had something similar recently but the issue was that I wasn't using coverage correctly, and it looks like coverage didn't wait for the test run to complete. 
How do you collect coverage?

**Comment by bergwerf** [comment]
With this script:
https://github.com/bergwerf/codecov-dart/blob/master/upload.sh

**Comment by zoechi** [comment]
The `wait` in there looks suspicious to me
I use
```
#!/usr/bin/env bash
pub global run coverage:collect_coverage --port=8111 --out=coverage/coverage.json --resume-isolates --wait-paused &
dart --observe=8111 test/all.dart
pub global run coverage:format_coverage --packages=.packages --in=coverage/coverage.json --lcov --report-on=lib --out=coverage/lcov.info
```
and didn't have any such issues since

**Comment by bergwerf** [comment]
I have updated the script and the same results seem to appear (unless
restarting a build in travis does not overwrite coverage results on codecov)

Op do 28 dec. 2017 om 10:27 schreef Günter Zöchbauer <
notifications@github.com>:

**Comment by rmacnak-google** [comment]
Likely fixed by 074b43c863f1b72a207f8c930ccda58980411d36
